.github/workflows/node-build.yml

```diff
@@ -159,6 +159,135 @@ jobs:
       - name: Run tests
         run: pnpm --filter integration run-tests
 
+  performance-test:
+    runs-on: ubuntu-22.04
+    needs: prerequisite
+    timeout-minutes: 5
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup environment
+        uses: ./.github/workflows/rafiki/env-setup
+
+      - name: Set up k6
+        uses: grafana/setup-k6-action@v1
+
+      - name: Setup hosts
+        run: |
+          echo "127.0.0.1 cloud-nine-wallet-test-backend" | sudo tee -a /etc/hosts
+          echo "127.0.0.1 cloud-nine-wallet-test-auth" | sudo tee -a /etc/hosts
+          echo "127.0.0.1 happy-life-bank-test-backend" | sudo tee -a /etc/hosts
+          echo "127.0.0.1 happy-life-bank-test-auth" | sudo tee -a /etc/hosts
+
+      - name: Build dependencies
+        run: pnpm --filter performance build:deps
+
+      - name: Bundle
+        run: pnpm --filter performance bundle
+
+      - name: Start test environment
+        run: pnpm --filter performance testenv:compose up --wait --build -d
+
+      - name: Start Mock Account Entity Services
+        run: nohup pnpm --filter performance start-mases > mases.log 2>&1 &
+
+      - name: Run performance tests
+        id: perf_test
+        run: |
+          pnpm --filter performance run-tests:testenv -k --vus 2 --duration 30s | tee performance.log
+        continue-on-error: true # Don't fail here, allows showing logs
+
+      - name: Print MASE logs
+        if: always()
+        run: cat mases.log
+
+      - name: Post/Update Performance Test Results as PR Comment
+        if: always()
+        uses: actions/github-script@v7
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
```

Comment:
I am in-lining the javascript here because I found it easier than passing everything from the actions context in and it was just easier for development. We can move this to a file and just run it with node if we prefer though.